server/src/main/java/org/elasticsearch/common/settings/RemoveSettingKeyStoreCommand.java

```
@@ -52,15 +53,23 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
         if (keystore == null) {
             throw new UserException(ExitCodes.DATA_ERROR, "Elasticsearch keystore not found. Use 'create' command to create one.");
         }
-
-        keystore.decrypt(new char[0] /* TODO: prompt for password when they are supported */);
-
-        for (String setting : arguments.values(options)) {
-            if (keystore.getSettingNames().contains(setting) == false) {
-                throw new UserException(ExitCodes.CONFIG, "Setting [" + setting + "] does not exist in the keystore.");
+        char[] passphrase = null;
+        try {
+            passphrase = keystore.hasPassword() ? KeyStoreWrapper.readPassphrase(terminal, false) : new char[0];
+            keystore.decrypt(passphrase);
+            for (String setting : arguments.values(options)) {
+                if (keystore.getSettingNames().contains(setting) == false) {
+                    throw new UserException(ExitCodes.CONFIG, "Setting [" + setting + "] does not exist in the keystore.");
+                }
+                keystore.remove(setting);
+            }
+            keystore.save(env.configFile(), passphrase);
+        } catch (SecurityException e) {
+            throw new UserException(ExitCodes.DATA_ERROR, "Failed to access the keystore. Please make sure the passphrase was correct.");
```

lets not swallow this exception; we should add them to the `UserException`